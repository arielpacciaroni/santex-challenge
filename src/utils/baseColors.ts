export const baseColors = {
  "light-gray": "#f5f5f5",
  "highlight": "#eee",
  "success": "#15803d",
  "secondary": '#1c1917',
};