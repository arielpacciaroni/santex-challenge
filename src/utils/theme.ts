const colors = {
  'light-gray': '#f5f5f5',
  highlight: '#eee',
  success: '#15803d',
  secondary: '#1c1917',
  information: '#bfdbfe',
  button: {
    'bg-primary': '#15803d',
    'bg-primary-hover': '#059669',
    'text-primary': '#dcfce7',
  },
};

const spacing = {
  small: '0.5rem',
  medium: '1rem',
};

export interface Theme {
  spacing: typeof spacing;
  colors: typeof colors;
}

export const theme = {
  spacing,
  colors,
};
