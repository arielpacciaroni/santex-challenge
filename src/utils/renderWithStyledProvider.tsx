import { render } from "@testing-library/react";
import { ReactNode } from "react"
import { ThemeProvider } from "styled-components";
import { theme } from "./theme";

export function renderWithStyledProvider(element: ReactNode) {
  return render(
    <ThemeProvider theme={theme}>
      {element}
    </ThemeProvider>
  )
}