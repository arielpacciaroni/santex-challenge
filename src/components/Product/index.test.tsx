import userEvent from '@testing-library/user-event';
import { Product } from '.';
import { renderWithStyledProvider } from '../../utils/renderWithStyledProvider';
import { fireEvent, screen } from '@testing-library/react';

describe('Product component', () => {
  it('should render a product', () => {
    const view = renderWithStyledProvider(
      <Product
        onAddToCart={jest.fn()}
        product={{
          id: '1',
          name: 'Product name',
          description: 'Product description',
          featuredAsset: {
            id: '1',
            preview: 'https://via.placeholder.com/300',
          },
          variants: [
            {
              id: '1',
              price: 100,
              name: 'Variant 1',
            },
          ],
        }}
      />
    );

    expect(screen.getByText('Product name')).toBeInTheDocument();
    expect(screen.getByText('Product description')).toBeInTheDocument();
    expect(screen.getByText('Choose a variant:')).toBeInTheDocument();
    expect(view).toMatchSnapshot();
  });

  it('should trigger the onAddToCart function when the button is clicked', () => {
    const onAddToCart = jest.fn();
    const variantId = 'someVariantId';
    renderWithStyledProvider(
      <Product
        onAddToCart={onAddToCart}
        product={{
          id: '1',
          name: 'Product name',
          description: 'Product description',
          featuredAsset: {
            id: '1',
            preview: 'https://via.placeholder.com/300',
          },
          variants: [
            {
              id: variantId,
              price: 100,
              name: 'Variant 1',
            },
          ],
        }}
      />
    );

    fireEvent.click(screen.getByText('Buy'));

    expect(onAddToCart).toHaveBeenCalledWith(variantId);
  });

  it('should change the selected variant when the select input changes', () => {
    const view = renderWithStyledProvider(
      <Product
        onAddToCart={jest.fn()}
        product={{
          id: '1',
          name: 'Product name',
          description: 'Product description',
          featuredAsset: {
            id: '1',
            preview: 'https://via.placeholder.com/300',
          },
          variants: [
            {
              id: '1',
              price: 100,
              name: 'Variant 1',
            },
            {
              id: '2',
              price: 200,
              name: 'Variant 2',
            },
          ],
        }}
      />
    );

    userEvent.selectOptions(
      screen.getByRole('combobox'),
      screen.getByRole('option', { name: 'Variant 2 - $200' })
    );

    expect(screen.getByText('Variant 2 - $200')).toBeInTheDocument();

    expect(view).toMatchSnapshot();
  });
});
