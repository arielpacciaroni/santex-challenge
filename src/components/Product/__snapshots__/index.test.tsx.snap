// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should change the selected variant when the select input changes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c7 {
  border: none;
  outline: none;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  font-size: 0.875rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-transition-delay: 100ms;
  transition-delay: 100ms;
  cursor: pointer;
  color: #dcfce7;
  background-color: #15803d;
}

.c7:hover {
  background-color: #059669;
}

.c7:disabled {
  background-color: #eee;
  color: #aaa;
  cursor: not-allowed;
}

.c5 {
  padding: 0.5rem;
  border-radius: 0.25rem;
  border: 1px solid #ccc;
  margin: 0.5rem 0;
  width: 100%;
  background-color: #fff;
  color: #000;
  cursor: pointer;
  -webkit-transition: border-color 0.2s ease-in-out;
  transition: border-color 0.2s ease-in-out;
}

.c5:focus {
  outline: none;
  border-color: #000;
}

.c0 {
  background-color: white;
  box-shadow: 0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1);
  border: 1px solid #ccc;
  border-radius: 0.5rem;
}

.c3 {
  margin: 0;
  font-size: 1.6rem;
  font-weight: bold;
}

.c2 {
  padding: 0.5rem;
}

.c1 {
  width: 100%;
  height: 200px;
  object-fit: cover;
  border-top-left-radius: 0.5rem;
  border-top-right-radius: 0.5rem;
}

.c4 {
  color: #666;
  margin: 1rem 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  margin: 0.5rem 0;
}

@media (min-width:768px) {
  .c1 {
    height: 300px;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <img
          alt="Product name"
          class="c1"
          src="https://via.placeholder.com/300"
        />
        <div
          class="c2"
        >
          <h2
            class="c3"
          >
            Product name
          </h2>
          <p
            class="c4"
          >
            Product description
          </p>
          <p>
            Choose a variant:
          </p>
          <select
            class="c5"
          >
            <option
              value="1"
            >
              Variant 1 - $100
            </option>
            <option
              value="2"
            >
              Variant 2 - $200
            </option>
          </select>
          <div
            class="c6"
          >
            <button
              class="c7"
            >
              Buy
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c7 {
  border: none;
  outline: none;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  font-size: 0.875rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-transition-delay: 100ms;
  transition-delay: 100ms;
  cursor: pointer;
  color: #dcfce7;
  background-color: #15803d;
}

.c7:hover {
  background-color: #059669;
}

.c7:disabled {
  background-color: #eee;
  color: #aaa;
  cursor: not-allowed;
}

.c5 {
  padding: 0.5rem;
  border-radius: 0.25rem;
  border: 1px solid #ccc;
  margin: 0.5rem 0;
  width: 100%;
  background-color: #fff;
  color: #000;
  cursor: pointer;
  -webkit-transition: border-color 0.2s ease-in-out;
  transition: border-color 0.2s ease-in-out;
}

.c5:focus {
  outline: none;
  border-color: #000;
}

.c0 {
  background-color: white;
  box-shadow: 0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1);
  border: 1px solid #ccc;
  border-radius: 0.5rem;
}

.c3 {
  margin: 0;
  font-size: 1.6rem;
  font-weight: bold;
}

.c2 {
  padding: 0.5rem;
}

.c1 {
  width: 100%;
  height: 200px;
  object-fit: cover;
  border-top-left-radius: 0.5rem;
  border-top-right-radius: 0.5rem;
}

.c4 {
  color: #666;
  margin: 1rem 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  margin: 0.5rem 0;
}

@media (min-width:768px) {
  .c1 {
    height: 300px;
  }
}

<div>
    <div
      class="c0"
    >
      <img
        alt="Product name"
        class="c1"
        src="https://via.placeholder.com/300"
      />
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Product name
        </h2>
        <p
          class="c4"
        >
          Product description
        </p>
        <p>
          Choose a variant:
        </p>
        <select
          class="c5"
        >
          <option
            value="1"
          >
            Variant 1 - $100
          </option>
          <option
            value="2"
          >
            Variant 2 - $200
          </option>
        </select>
        <div
          class="c6"
        >
          <button
            class="c7"
          >
            Buy
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Product component should render a product 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c7 {
  border: none;
  outline: none;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  font-size: 0.875rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-transition-delay: 100ms;
  transition-delay: 100ms;
  cursor: pointer;
  color: #dcfce7;
  background-color: #15803d;
}

.c7:hover {
  background-color: #059669;
}

.c7:disabled {
  background-color: #eee;
  color: #aaa;
  cursor: not-allowed;
}

.c5 {
  padding: 0.5rem;
  border-radius: 0.25rem;
  border: 1px solid #ccc;
  margin: 0.5rem 0;
  width: 100%;
  background-color: #fff;
  color: #000;
  cursor: pointer;
  -webkit-transition: border-color 0.2s ease-in-out;
  transition: border-color 0.2s ease-in-out;
}

.c5:focus {
  outline: none;
  border-color: #000;
}

.c0 {
  background-color: white;
  box-shadow: 0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1);
  border: 1px solid #ccc;
  border-radius: 0.5rem;
}

.c3 {
  margin: 0;
  font-size: 1.6rem;
  font-weight: bold;
}

.c2 {
  padding: 0.5rem;
}

.c1 {
  width: 100%;
  height: 200px;
  object-fit: cover;
  border-top-left-radius: 0.5rem;
  border-top-right-radius: 0.5rem;
}

.c4 {
  color: #666;
  margin: 1rem 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  margin: 0.5rem 0;
}

@media (min-width:768px) {
  .c1 {
    height: 300px;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <img
          alt="Product name"
          class="c1"
          src="https://via.placeholder.com/300"
        />
        <div
          class="c2"
        >
          <h2
            class="c3"
          >
            Product name
          </h2>
          <p
            class="c4"
          >
            Product description
          </p>
          <p>
            Choose a variant:
          </p>
          <select
            class="c5"
          >
            <option
              value="1"
            >
              Variant 1 - $100
            </option>
          </select>
          <div
            class="c6"
          >
            <button
              class="c7"
            >
              Buy
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c7 {
  border: none;
  outline: none;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  font-size: 0.875rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-transition-delay: 100ms;
  transition-delay: 100ms;
  cursor: pointer;
  color: #dcfce7;
  background-color: #15803d;
}

.c7:hover {
  background-color: #059669;
}

.c7:disabled {
  background-color: #eee;
  color: #aaa;
  cursor: not-allowed;
}

.c5 {
  padding: 0.5rem;
  border-radius: 0.25rem;
  border: 1px solid #ccc;
  margin: 0.5rem 0;
  width: 100%;
  background-color: #fff;
  color: #000;
  cursor: pointer;
  -webkit-transition: border-color 0.2s ease-in-out;
  transition: border-color 0.2s ease-in-out;
}

.c5:focus {
  outline: none;
  border-color: #000;
}

.c0 {
  background-color: white;
  box-shadow: 0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1);
  border: 1px solid #ccc;
  border-radius: 0.5rem;
}

.c3 {
  margin: 0;
  font-size: 1.6rem;
  font-weight: bold;
}

.c2 {
  padding: 0.5rem;
}

.c1 {
  width: 100%;
  height: 200px;
  object-fit: cover;
  border-top-left-radius: 0.5rem;
  border-top-right-radius: 0.5rem;
}

.c4 {
  color: #666;
  margin: 1rem 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  margin: 0.5rem 0;
}

@media (min-width:768px) {
  .c1 {
    height: 300px;
  }
}

<div>
    <div
      class="c0"
    >
      <img
        alt="Product name"
        class="c1"
        src="https://via.placeholder.com/300"
      />
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Product name
        </h2>
        <p
          class="c4"
        >
          Product description
        </p>
        <p>
          Choose a variant:
        </p>
        <select
          class="c5"
        >
          <option
            value="1"
          >
            Variant 1 - $100
          </option>
        </select>
        <div
          class="c6"
        >
          <button
            class="c7"
          >
            Buy
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
